src/emmy/structure.cljc

```diff
@@ -489,9 +532,14 @@
 
 (defn make
   "Generate a structure with the supplied orientation, given some sequence `xs`"
-  [orientation xs]
-  (let [xs (if (vector? xs) xs (into [] xs))]
-    (->Structure orientation xs nil)))
+  ([orientation xs]
+   (make orientation xs nil))
+  ([orientation xs meta]
+   (let [xs (if (vector? xs) xs (into [] xs))]
+     (->Structure orientation xs meta)
+     #_(-> (->Structure orientation xs meta)
```

Comment:
can we remove this if it's not used? Or do we need this?

Reply:
I toyed with the idea of forcing all structure creation through this point. There is still a debate to be had about that. Making an object a Function that wasn't born that way requires the use of `Object.setPrototypeOf` which is [strongly discouraged](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/setPrototypeOf) at MDN since it frustrates certain optimizations. 

I found that Markus' tests could be done without medicine this strong, by moving these calls to the TypeScript tuple constructors. The possibility remains (and indeed there is one such case among those tests) in which an up-tuple is generated by interior code below the TypeScript layer but pops back out in user land without this decoration. This is rare though.

The decision of whether it is better to treat all structs the same even at the risk of annoying V8 or not is one we might use a spreadsheet for. One intermediate example is to have a *dynamic-var* that controls this behavior, and have the `emmy.js` library opt into this feature by setting that var true. 

I'm trying to protect the performance/principle of what we have (though I don't know if this is any slower). The argument on the other side is "Emmy is a dialect of the host language, and the transparency of its use is going to require things like this; brace yourself."